File: backend/app/models/post.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base


class Post(Base):
    __tablename__ = "posts"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    category_id = Column(Integer, ForeignKey("categories.id", ondelete="SET NULL"), nullable=True, index=True)
    title = Column(String(200), nullable=False)
    content = Column(Text, nullable=False)
    image_url = Column(String(500), nullable=True)
    view_count = Column(Integer, nullable=False, default=0, server_default="0")
    like_count = Column(Integer, nullable=False, default=0, server_default="0")
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    user = relationship("User", back_populates="posts")
    category = relationship("Category", back_populates="posts")
    comments = relationship("Comment", back_populates="post", cascade="all, delete-orphan")
    reactions = relationship("Reaction", back_populates="post", cascade="all, delete-orphan")

File: frontend/src/types/post.ts
export interface Post {
  id: number;
  user_id: number;
  category_id: number | null;
  title: string;
  content: string;
  image_url: string | null;
  image_urls: string[];
  is_anonymous: boolean;
  view_count: number;
  like_count: number;
  created_at: string;
  updated_at: string;
}

export interface PostImage {
  id: number;
  post_id: number;
  url: string;
  sort_order: number;
  created_at: string;
}

export interface PostImageListResponse {
  images: PostImage[];
}

export interface PostCreate {
  category_id?: number | null;
  title: string;
  content: string;
  image_url?: string | null;
  image_urls?: string[];
  is_anonymous?: boolean;
}

export interface PostUpdate {
  category_id?: number | null;
  title?: string;
  content?: string;
  image_url?: string | null;
  image_urls?: string[];
  is_anonymous?: boolean;
}

export interface PostListResponse {
  posts: Post[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}
